System/Secretary/DeskEventManager.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * Manages event handlers for an event aware class
+ */
+import { DeskEvent } from "./DeskEvent";
+
+interface DeskEventItem {
+    id: string;
+    deskEvent: DeskEvent;
+}
+
+class DeskEventInfo {
+    id: string;
+    constructor(id: string) {
+        this.id = id;
+    }
+}
+
+export class DeskEventManager {
+    /**
+     * DeskEvent array
+     */
+    private deskEventItems: DeskEventItem[] = [];
+    private lastIndex: -1;
+
+    add(target: Element | Document | Window, method: string, evtFunc: (this: Element, ev: any) => any): DeskEventInfo {
+        const deskEvent = this.deskEventItems.find(
+            (x) => x.deskEvent.target === target && x.deskEvent.method === method && x.deskEvent.evtFunc === evtFunc
+        );
+        if (deskEvent) {
+            return new DeskEventInfo(deskEvent.id);
+        }
```

Comment:
I think this logic is too smart/unexpected for an add function. I'd consider just doing the expected thing and dumbly adding another event. What do you think @qlqhwb?